src/libos/crates/async-rt/src/lib.rs

```diff
@@ -199,6 +206,195 @@ mod tests {
         .spawn()
     }
 
+    #[test]
+    fn test_mutex() {
```

Comment:
Unit tests for mutexes should be moved to mutex module. So are rwlock tests. We cannot write all unit tests in `lib.rs`.  Most unit tests in `lib.rs` are there only for historical reasons; they should be put under appropriate modules, instead of the root modules.